### ISSUE TITLE
Text not rendered correctly as an image with the packaged `libgd3` version

### ISSUE THREAD

**Issue by trantor** [oerdnj/deb.sury.org#2236]
**Frequently asked questions**

- [x] I have read [Frequently Asked Questions](https://github.com/oerdnj/deb.sury.org/wiki/Frequently-Asked-Questions)
- [x] I have looked at the list of the existing issues (including closed issues) and searched if my issue has been already reported
- [x] I have tried to resolve the issue myself and will describe what I did in clear and concise manner

**Describe the bug**
I run an instance of [Zabbix](https://www.zabbix.com/) whose web interface runs on a local PHP 8.3 (FPM) installation sourced from the Ubuntu PPA.
After upgrading to Ubuntu 24.04.1 and PHP 8.3 we noticed a problem in the image rendering of graphs, which is accomplished by the web interface through the PHP GD extension.

A normal rendering would look something like this

![image](https://github.com/user-attachments/assets/697c4196-a6ca-457a-b3a7-3824c6e34083)

What we would get instead is 
![image](https://github.com/user-attachments/assets/b055a677-4c81-4b0d-a4ca-a5c140d1d0ec)

Going through the code I tracked the problem to a [imagettftext](https://www.php.net/manual/it/function.imagettftext.php) call at https://git.zabbix.com/projects/ZBX/repos/zabbix/browse/ui/include/graphs.inc.php#476

The text rendered as an image appears to be squashed horizontally.

I then tried bringing the `libgd3` package for the [GD library](https://libgd.github.io/) back from version `2.3.3-12+ubuntu24.04.1+deb.sury.org+1` to the distro stock version `2.3.3-9ubuntu5` and restarted the `php8.3-fpm` service.
And the text rendering went back to normal.

Another thing is that installing the PPA version installed as depedencies the libraries below, which get uninstalled by switching back to the stock Ubuntu version of `libgd3`

```
Setting up libgraphite2-3:arm64 (1.3.14-2build1) ...
Setting up libsvtav1enc1d1:arm64 (1.7.0+dfsg-2build1) ...
Setting up librav1e0:arm64 (0.7.1-2) ...
Setting up libimagequant0:arm64 (2.18.0-1build1) ...
Setting up libdav1d7:arm64 (1.4.1-1build1) ...
Setting up libharfbuzz0b:arm64 (8.3.0-2build2) ...
Setting up libabsl20220623t64:arm64 (20220623.1-3.1ubuntu3) ...
Setting up libyuv0:arm64 (0.0~git202401110.af6ac82-1) ...
Setting up libraqm0:arm64 (0.10.1-1build1) ...
Setting up libgav1-1:arm64 (0.18.0-1build3) ...
Setting up libavif16:arm64 (1.0.4-1ubuntu3) ...
Setting up libgd3:arm64 (2.3.3-12+ubuntu24.04.1+deb.sury.org+1) ...
```

This is also _likely_ the cause of the similar-looking https://support.zabbix.com/browse/ZBX-25191

**To Reproduce**
Hard to provide a simple way. Probably some simple PHP code using the `imagettftext` function would do the trick.

**Your understanding of what is happening**
Something with the patches to the gd library code to compile this package is causing text rendered as an image to be squashed horizontally.

**What steps did you take to resolve issue yourself before reporting it here**
I looked through the stack of the image rendering until I arrived to the underlying GD library. Substituting it with a "stock" version solved the problem, hence the packaged version is likely the problem

**Expected behavior**
For text to be rendered legibly as an image through the PHP GD extension and the underlying GD library

**Distribution (please complete the following information):**
 - OS: Ubuntu 24.04.1
 - Architecture: arm64
 - Repository: Ubuntu PPA

**Package(s) (please complete the following information):**
```
libgd3:
  Installed: 2.3.3-9ubuntu5
  Candidate: 2.3.3-12+ubuntu24.04.1+deb.sury.org+1
  Version table:
     2.3.3-12+ubuntu24.04.1+deb.sury.org+1 500
        500 http://ppa.launchpad.net/ondrej/php/ubuntu noble/main arm64 Packages
 *** 2.3.3-9ubuntu5 500
        500 http://eu-west-1.ec2.ports.ubuntu.com/ubuntu-ports noble/main arm64 Packages
        100 /var/lib/dpkg/status
```
The listing above is after switching back to the stock Ubuntu package

**Additional context**
I am unsure whether the `arm64` architecture might have an impact on the issue.

**Comment by selivan** [comment]
I confirm this bug. Software versions:
* OS: ubuntu 24.04
* libgd3:arm64 `2.3.3-12+ubuntu24.04.1+deb.sury.org+1`
* zabbix-frontend-php `7.0.6`

**Comment by trantor** [comment]
Actually, the patchset seems to be identical to the one from the official Ubuntu distribution. What changes are the dependencies. The Ubuntu package is not built with dependencies on libramq, libavif, libimagequant .

**Comment by oerdnj** [comment]
My guess is that the bug is somewhere in the support for rendering RTL languages. But I would need a reproducer to see what should be disabled or fixed. Perhaps filling this against upstream libgd might also help? Someone might take a look, but reproducer would be really useful.

**Comment by trantor** [comment]
@oerdnj yep, that would make sense to me as well. Although this issue https://github.com/libgd/libgd/issues/887 seems to fit the bill quite well.

**Comment by trantor** [comment]
@oerdnj in the issue mentioned above they seem to imply that the `libgd` code was fixed in 2022 but the debian package has not incorporated the fix.

**Comment by zaidpirwani** [comment]
> I then tried bringing the `libgd3` package for the [GD library](https://libgd.github.io/) back from version `2.3.3-12+ubuntu24.04.1+deb.sury.org+1` to the distro stock version `2.3.3-9ubuntu5` and restarted the `php8.3-fpm` service.
> And the text rendering went back to normal.


I was having the same issue, I installed the older version as per comment and my issue is resolved


apt-cache policy libgd3
libgd3:
  Installed: 2.3.3-9ubuntu5
  Candidate: 2.3.3-12+ubuntu24.04.1+deb.sury.org+1
  Version table:
     2.3.3-12+ubuntu24.04.1+deb.sury.org+1 500
        500 https://ppa.launchpadcontent.net/ondrej/php/ubuntu noble/main amd64 Packages
 *** 2.3.3-9ubuntu5 500
        500 http://ap-southeast-1.ec2.archive.ubuntu.com/ubuntu noble/main amd64 Packages
        100 /var/lib/dpkg/status

sudo apt install libgd3=2.3.3-9ubuntu5

Ubuntu 24.04.2 LTS, PHP8.3, Zabbix 7.2.4

**Comment by DronKram** [comment]
To pin the libgd3 version 2.3.3-9ubuntu5 you can create a file libgd3 in the /etc/apt/preferences.d/ (/etc/apt/preferences.d/libgd3) and insert the next lines into it

Package: libgd3
Pin: version 2.3.3-9*
Pin-Priority: 600

This will up the priority for this version and an update will not be installed during apt upgrade.

**Comment by oerdnj** [comment]
I've removed the libraqm support from the src:libgd2 package
